### PR TITLE
don't always gunzip layers

### PR DESCRIPTION
**Pull request by tych0**
Layers can be stored in tar format alone, so let's not force gunzip.

Signed-off-by: Tycho Andersen <tycho@tycho.ws>

### CODE REVIEW

File: oci/layer/unpack.go [hunk 1]
```diff
@@ -226,7 +234,7 @@ func UnpackManifest(ctx context.Context, engine cas.Engine, bundle string, manif
 		// in the layer.
 		_, _ = io.Copy(ioutil.Discard, layer)
 		// XXX: Is it possible this breaks in the error path?
-		layerGzip.Close()
+		layerRaw.Close()
```
Comment (cyphar) [inline oci/layer/unpack.go]: This should be `layerData`, assuming you wanted to have the same semantics.

Comment (tych0) [reply oci/layer/unpack.go]: Arg, yeah, thanks. Just pushed a fix.

**Comment by cyphar** [comment]
LGTM.

**Comment by cyphar** [comment]
Sorry, something's wrong with my `lgtm.co` instance. I'm looking into it...

**Comment by cyphar** [comment]
LGTM.

**Pull request merged**